game-core/src/main/java/games/strategy/triplea/attachments/UnitSupportAttachment.java

```diff
@@ -60,7 +63,8 @@
     private static final long serialVersionUID = -7445551357956238314L;
 
     @Nonnull String name;
-    @Nonnull Integer count;
+
+    @EqualsAndHashCode.Exclude @Nonnull Integer count;
```

Comment:
What's the rationale for excluding 'count' from equals and hash code?

Reply:
The original code used the name as the equality.  And, per pact of steel, "Any support of the same bonusType MUST have the same count.".

Since I'm changing the code to use the object as the hash key instead of just the name of the bonus, I excluded the count to prevent an accidental bug where a map has different counts for the same bonusType.

I checked the maps and there is one map that has different counts for the same bonusType: warcraft war heroes.  The "Wall-Archer" bonusType has a count of 1 and a count of 5.  The game is currently picking the first one in the list, which might be 1 or it might be 5.